ports/stm32/mpmetalport.h

```diff
@@ -70,7 +70,7 @@ static inline int __metal_sleep_usec(unsigned int usec) {
 }
 
 static inline void metal_generic_default_poll(void) {
-    MICROPY_EVENT_POLL_HOOK
+    mp_event_handle_nowait();
```

Comment:
This changes the semantics of the poll: previously it would do a WFI, now it doesn't.

Where exactly is this function used?  I can see some code in the libmetal submodule (function called `metal_condition_wait`) that call this poll function with IRQs disabled.  Is that ever actually used?

Note that if IRQs are disabled then a WFE may never return (but a WFI will).

Reply:
Yes you have a point, but there's no `mp_event` that uses WFI, did I miss it ? This function is called from libmetal, why/where specifically I'm not really sure, but we need to provide an implementation.

Reply:
The equivalent would be:
```
mp_event_handle_nowait();
WFI();
```

Reply:
Fixed.